securedrop/journalist_app/api2/__init__.py

```diff
@@ -0,0 +1,137 @@
+import hashlib
+from typing import Any, Dict
+
+from flask import Blueprint, abort, json, jsonify, request
+from models import Source
+from sqlalchemy.orm import joinedload
+from werkzeug.wrappers.response import Response
+
+blp = Blueprint("api2", __name__, url_prefix="/api/v2")
+
+
+def json_version(d: dict) -> str:
+    """
+    Calculate the version (BLAKE2s digest) of the normalized JSON representation
+    of the dictionary ``d``.
+
+    We use BLAKE2s here because SHA-256 is too slow (we don't care about
+    cryptographic security) and CRC-32 is too collision-prone (we're not merely
+    checksumming for transmission integrity).
+    """
+    s = json.dumps(d, sort_keys=True)
+    b = s.encode("utf-8")
+    return hashlib.blake2s(b).hexdigest()
+
+
+@blp.get("/index")
+def index() -> Response:
+    """
+    Return the index of all sources.
```

Comment:
Nit: In #7579, this was documented in reference to a preceding Marshmallow schema.  Without that, let's be more explicit here:

```suggestion
    Return the `{uuid: version}` index of all sources.
```